app/code/Meta/Conversion/view/frontend/templates/pixel/add_to_cart.phtml

```diff
@@ -37,71 +37,72 @@ $trackerUrl = $block->getTrackerUrl();
+                            simpleProductId = configurableProduct.simpleProduct;
                         }
                     }
-                    // otherwise use the product sku
-                    const product_sku = data.sku;
-                    // use last added product id if swatch data is not available
-                    if (!simpleProductId) {
-                        simpleProductId = productId;
-                    }
-                    const form_key = jQuery("[name='form_key']").val();
-                    $.ajax({
-                        url: product_info_url,
-                        data: {
-                            product_sku: product_sku,
-                            product_id: simpleProductId,
-                            form_key: form_key
-                        },
-                        type: 'get',
-                        dataType: 'json',
-                        async: !data.response.backUrl,
-                        success: function (res) {
-                            const addToCartConfigPixel = {
-                                "url": window.addToCartData.url,
-                                "payload": {
-                                    "eventName": window.addToCartData.eventName,
-                                    "productId": res.productId
-                                },
-                                "browserEventData": {
-                                    'fbAgentVersion': window.addToCartData.fbAgentVersion,
-                                    'fbPixelId': window.addToCartData.fbPixelId,
-                                    'source': window.addToCartData.source,
-                                    'pluginVersion': window.addToCartData.pluginVersion,
-                                    'track': window.addToCartData.track,
-                                    'event': window.addToCartData.event,
-                                    'payload': {
-                                        "content_name": res.name,
-                                        "content_ids": [res.id],
-                                        "value": res.value,
-                                        "currency": res.currency,
-                                        "content_type": res.content_type,
-                                        "contents": [{
-                                            "id": res.id,
-                                            "quantity": 1
-                                        }]
-                                    }
+                }
+                // otherwise use the product sku
+                const product_sku = data.sku;
+                // use last added product id if swatch data is not available
+                if (!simpleProductId) {
+                    simpleProductId = productId;
+                }
+                const form_key = jQuery("[name='form_key']").val();
+                const runAsync = !data.response.backUrl;
```

Comment:
main addition here

Reply:
this will still result in an error if `data.response.backUrl` is `undefined`. add a check here, something like:

`const runAsync = 'backUrl' in data.response ? !data.response.backUrl : true;`

– returns `true` if `data.response.backUrl` is `undefined` or `false`

Reply:
This is redundant because undefined checks are meant to be used for falsy value checking. To your point I added `cont runAsync = !(data && data.response && data.response.backUrl)` to cover our bases but explicitly checking for undefined like you have here is not only redundant but misses a case: backUrl could be set to empty string for example and that is functionally the same as it not being set at all. Just checking it the way I have it written would handle both cases.
<img width="1119" alt="Screenshot 2024-12-09 at 5 07 39 PM" 
<img width="669" alt="image" src="https://github.com/user-attachments/assets/b35d3dfb-c4e0-4fc7-b06c-acc72cf4fb04">
<img width="669" alt="Screenshot 2024-12-09 at 5 42 02 PM" src="https://github.com/user-attachments/assets/148cdc1f-8d44-4e60-b66b-422e59ee693c">
src="https://github.com/user-attachments/assets/af0ce705-1fbe-4f2f-bdb9-b839c9700522">

Reply:
<img width="1119" alt="Screenshot 2024-12-09 at 5 07 39 PM" src="https://github.com/user-attachments/assets/bd533599-3f13-49c7-a42e-ef306cdce271">